src/sidebars/favourites/favourite-controller.js

```diff
@@ -100,7 +100,8 @@
                 if (updatedStock) {
                     if (this.stocks.length === 0) {
                         // If there aren't any stocks, we could be adding one...
-                        if (updatedStock.favourite) {
+                        // But only if there's no selection already
+                        if (updatedStock.favourite && !this.selection()) {
```

Comment:
This is relying on the empty object in the selection service, right? So also relying on the behaviour of `!` with `''`? If so that might not be obvious for reading/debugging in the future.